docs/CppInteroperabilityManifesto.md

```diff
@@ -2575,6 +2576,45 @@ We could ignore explicit specializations of function templates, because they
 don't affect the API. Explicit specializations of class templates can
 dramatically change the API of the type.
 
+### Class templates: Importing full class template instantiations
+
+A class template instantiation could be imported as a struct named
+`__CxxTemplateInst` plus Itanium mangled type of the instantiation (see the
```

Comment:
Instead of prefixing each declaration name, could we add another special module context like `So` and `SC` for ObjC/C synthesized declarations?

Reply:
Yes we could. The benefit would be having shorter mangled names and smaller metadata. The downside is slight increase in implementation complexity. For my use cases, I'd definitely appreciate any shortening of mangled names I can get. However, we biased towards reducing implementation complexity.

What's your preference?